src/game/entities/obstacle/Obstacle.ts

```diff
@@ -1,22 +1,43 @@
-import { GAME_SETTINGS } from 'game/constants';
+import { GAME_SETTINGS, OBSTACLE_TYPES } from 'game/constants';
 import { Entity, Sprite } from 'game/core';
-import { TCoordinates } from 'game/core/types';
+import { TCoordinates, TSize, TSpeedModifierRefernce } from 'game/core/types';
 
 export default class Obstacle extends Entity {
-  constructor(initialPosition: TCoordinates) {
-    super(
+  constructor(
+    initialPosition: TCoordinates,
+    speedModifierRefernce?: TSpeedModifierRefernce,
+    obstacleType: string = OBSTACLE_TYPES.TYPE1.name
+  ) {
+    let startCoordinates: TCoordinates = { x: 0, y: 133 };
+    let size: TSize = { width: 187, height: 84 };
+
+    if (obstacleType) {
+      switch (obstacleType) {
+        case OBSTACLE_TYPES.TYPE2.name:
+          startCoordinates = { x: 188, y: 197 };
+          size = { width: 187, height: 18 };
+          break;
```

Comment:
🧐
Если бы размеры и полноценные координаты хранились в константе, то от свитча можно было бы избавиться, и записать это как-то так:
```suggestion
        startCooridnates = OBSTACLE_TYPES[obstacleType].coords;
        size = OBSTACLE_TYPES[obstacleType].size;
```

Это хорошо ложится на стратегию выноса информации об игровых объектах и показателях в константы. Не помечаю как обязательную правку, но очень nice-to-have